### ISSUE TITLE
 [Feature Request] Adding clip_grap_value and supporting sparse gradients

### ISSUE THREAD

**Issue by berzjackson** [pytorch/pytorch#4829]
1.It would be great to have another common gradient clipping strategy, i.e. clip_grad_value in PyTorch. Compared to clip_grad_norm, which is sensitive to the total number of parameters in the model, clip_grad_value will be easier to use as tuning max_value will be more straightforward than tuning max_norm. Still, AFAIK clip_grad_norm is the recommended way to do gradient clipping since it preserves the direction of the gradients while clip_grad_value does not.

2.The current implementation of clip_grad_norm can not handle sparse gradients. A common case use case is when sparse=True in nn.Embedding layers.

I will issue a PR if people feel this is worth adding.

**Comment by apaszke** [comment]
1. I don't think this needs a PR, since it can be achieved by:
```python
for p in model.parameters():
    p.grad.clamp(max=max_grad_val)
```

2. Sounds good.

**Comment by soumith** [comment]
(2) is handled in #4882 which is merged now

**Comment by zou3519** [comment]
What's left to do for (2) is to possibly implement `abs` for sparse tensors or find a workaround for the inf norm. If you're not using the inf norm then it should work as advertised.

**Comment by mruberry** [comment]
Updating title to reflect remaining work.
